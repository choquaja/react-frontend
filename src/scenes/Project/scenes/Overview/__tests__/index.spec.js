import React from 'react';
import { shallow } from 'enzyme';
import { Overview } from '../';

describe('Overview', () => {
  it('renders without crashing', () => {
    const props = {
      loading: false,
    };
    shallow(<Overview {...props} />);
  });

  it('renders loading indicator', () => {
    const props = {
      loading: true,
      data: null,
    };
    const component = shallow(<Overview {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('renders empty content message', () => {
    const props = {
      loading: false,
      data: null,
    };
    const component = shallow(<Overview {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('renders data regardless of loading', () => {
    const props = {
      data: { content: '# Sample Title' },
    };
    const component1 = shallow(<Overview {...props} loading={false} />);
    expect(component1).toMatchSnapshot();
    const component2 = shallow(<Overview {...props} loading />);
    expect(component2).toMatchSnapshot();
  });
});
