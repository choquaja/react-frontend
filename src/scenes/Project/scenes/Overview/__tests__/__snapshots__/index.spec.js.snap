// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview renders data regardless of loading 1`] = `
<AnimFade>
  <ContentCard__ContentCard
    column={true}
  >
    <CardTitle__CardTitle
      invert={false}
    >
      Overview
    </CardTitle__CardTitle>
    <p>
      # Sample Title
    </p>
  </ContentCard__ContentCard>
</AnimFade>
`;

exports[`Overview renders data regardless of loading 2`] = `
<AnimFade>
  <ContentCard__ContentCard
    column={true}
  >
    <CardTitle__CardTitle
      invert={false}
    >
      Overview
    </CardTitle__CardTitle>
    <p>
      # Sample Title
    </p>
  </ContentCard__ContentCard>
</AnimFade>
`;

exports[`Overview renders empty content message 1`] = `
<AnimFade>
  <ContentCard__ContentCard
    column={true}
  >
    <CardTitle__CardTitle
      invert={false}
    >
      Overview
    </CardTitle__CardTitle>
    <NoContent>
      You're project doesn't contain a README.md file.
      <br />
      Why don't you 
      <a
        href="#empty"
      >
        create one?
      </a>
    </NoContent>
  </ContentCard__ContentCard>
</AnimFade>
`;

exports[`Overview renders loading indicator 1`] = `
<LoadingIndicator
  size={128}
/>
`;
